File: 2015/Day-1/Day1.py
f = open("input.txt", "r")

data = [x.strip() for x in f.read()]

f.close()

# Part 1
# File of parens dictating amount traversed
# ( = Up one floor (addition)
# ) = Down one floor (subtraction)

def Part1(data):
   # Begin Part 1 of Advent of Code 2015

   total = 0
   for step in data:
       if step == '(':
           total += 1
       else:
           total -= 1
   return total

print("Solution to Part 1: %d" % Part1(data))

# Part 2
# Same rules as above
# When on floor 0 (base floor), another ) will lead to the basement (-1)
# Determine which step in the process leads to the basement

def Part2(data):
    # Begin Part 2 of Advent of Code 2015

    position = 1
    total = 0
    for step in data:
        if step == '(':
            total += 1
        else:
            total -= 1
            if total == -1:
                return(position)
        position += 1

print("Solution to Part 2: %d" % Part2(data))

File: 2015/Day-1/Day1.lua
-- Import file
file = io.open("input.txt", "r")
data = file:read("*a")
file:close()

-- Step 1
p1Total = 0
for i = 1, #data do
  if data:sub(i,i) == "(" then p1Total = p1Total + 1 else p1Total = p1Total - 1 end
end

print(p1Total)
-- Step 2
p2Total = 0;
p2Tally = 1;

for i = 1, #data do
  if data:sub(i,i) == "(" then p2Total = p2Total + 1 else p2Total = p2Total - 1 end
  if p2Total == -1 then
    p2Tally = i
    break
  end
end

print(p2Tally)